src/output-model.ts

```diff
@@ -0,0 +1,96 @@
+/// <reference path="./typings/events.d.ts" />
+
+import {IOutput} from './i-output';
+import {IOutputModel} from './i-output-model';
+import {EventEmitter} from 'events';
+
+/**
+ * Provides Jupyter outputs to an output area.
+ *
+ * This class manages the conversion of Jupyter messages to output state, and
+ * handles adding the output state to the output area.
+ */
+export class OutputModel extends EventEmitter implements IOutputModel {
+    private _clear_queued: boolean = false;
+    private _state: IOutput[];
+    
+    /**
+     * Public constructor
+     */
+    public constructor() {
+        super();
+        this._state = [];
+    }
+    
+    /**
+     * State
+     * @return {IOutput[]}
+     */
+    public get state(): IOutput[] {
+        return this._state.slice();
+    }
+    /**
+     * State
+     * @param  {IOutput[]} value new state
+     */
+    public set state(value: IOutput[]) {
+        this.emit('change', value, this._state);
+        this._state = value;
+    }
+    
+    /**
+     * Consumes a Jupyter msg protocol message.
+     * Ignores messages that it doesn't know how to handle.
+     * @param  {any}     msg Jupyter protocol msg JSON
+     * @return {boolean}     was the msg consumed
+     */
+    public consume_msg(msg: any): boolean {
+        var state: any[] = this.state;
```

Comment:
Good catch- I'll have to address this.  My original intent was to only allow the underlying state to modified through a single entry point, so I could do complex things with events (right now I'm only throwing a change event).  

I'm still not sure how I want to implement the state, after talking to Brian and @Carreau I think I need to emit a finer grain of event.  Alternatively, I can add the fine grained events in another model implementation, maybe `rt-output-model`.  Now that I think of it, that may be a better idea...